packages/angular/src/migrations/update-15-7-0/install-required-packages.ts

```diff
@@ -18,13 +18,13 @@ export default async function (tree: Tree) {
   );
 
   const angularCliVersion =
-    pkgJson.devDependencies['@angular/cli'] ??
-    pkgJson.dependencies['@angular/cli'] ??
+    pkgJson.devDependencies?.['@angular/cli'] ??
+    pkgJson.dependencies?.['@angular/cli'] ??
     angularDevkitVersion;
 
   const filteredPackages: Record<string, string> = packagesToInstall
     .filter(
-      (pkg) => !pkgJson.devDependencies[pkg] && !pkgJson.dependencies[pkg]
+      (pkg) => !pkgJson.devDependencies?.[pkg] && !pkgJson.dependencies?.[pkg]
     )
     .reduce((allPkgs, pkg) => ({ ...allPkgs, [pkg]: angularCliVersion }), {});
```

Comment:
this migration could use the new util method you added

Reply:
Yeah, but I prefer not having the migration depend on a generator internal utility. The generator needs could change over time, but migrations are more like snapshots. For specific versions, they need a specific implementation.